test/GPv2Settlement/OrderRefunds.t.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {Helper} from "./Helper.sol";
+import {Order} from "test/libraries/Order.sol";
+
+enum FreeFunctionVariant {
+    FreeFilledAmountStorage,
+    FreePreSignatureStorage
+}
+
+abstract contract Variant is Helper {
+    FreeFunctionVariant internal freeFn;
+
+    constructor(FreeFunctionVariant _freeFn) {
+        freeFn = _freeFn;
+    }
+
+    function defaultOrderUids() internal view returns (bytes[] memory orderUids) {
+        orderUids = new bytes[](3);
+        orderUids[0] = Order.computeOrderUid(keccak256("order1"), trader.addr, 0);
+        orderUids[1] = Order.computeOrderUid(keccak256("order2"), trader.addr, 0);
+        orderUids[2] = Order.computeOrderUid(keccak256("order3"), trader.addr, 0);
+    }
+
+    function freeFunctionCall(bytes[] memory orderUids) private {
+        if (freeFn == FreeFunctionVariant.FreeFilledAmountStorage) {
+            settlement.freeFilledAmountStorage(orderUids);
+        } else if (freeFn == FreeFunctionVariant.FreePreSignatureStorage) {
+            settlement.freePreSignatureStorage(orderUids);
+        } else {
+            revert("Invalid free function");
+        }
+    }
+
+    function freeFunctionCallTest(bytes[] memory orderUids) private {
+        if (freeFn == FreeFunctionVariant.FreeFilledAmountStorage) {
+            settlement.freeFilledAmountStorageTest(orderUids);
+        } else if (freeFn == FreeFunctionVariant.FreePreSignatureStorage) {
+            settlement.freePreSignatureStorageTest(orderUids);
+        } else {
+            revert("Invalid free function");
+        }
+    }
+
+    function test_should_revert_if_not_called_from_an_interaction() public {
+        bytes[] memory emptyOrderUids = new bytes[](0);
+
+        vm.expectRevert("GPv2: not an interaction");
+        freeFunctionCall(emptyOrderUids);
+    }
+
+    function test_should_revert_if_the_encoded_order_uid_are_malformed() public {
+        bytes memory orderUidLt =
+            hex"00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
+        bytes memory orderUidGt =
+            hex"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
```

Comment:
Readability nit, so it's easy to check how long the bytestring is. Needs importing `GPv2Order` first:

```
import {GPv2Order} from "src/contracts/libraries/GPv2Order.sol";
```

```suggestion
        bytes memory orderUidLt = new bytes(GPv2Order.UID_LENGTH - 1);
        bytes memory orderUidGt = new bytes(GPv2Order.UID_LENGTH + 1);
```